src/audio/chain_dma.c

```diff
@@ -338,10 +338,12 @@ static int chain_task_pause(struct comp_dev *dev)
 	if (!ret)
 		ret = ret2;
 
+	k_spin_unlock(&drivers->lock, key);
+
```

Comment:
Note: the chain_task_pause() is not called in parallel with chain_task_start() so this lock does not seem to be needed. A follow-up PR might be needed to refactor the locking. This PR focuses to fix a a specific regression when asserts are enabled.